### PR TITLE
TRUNK-1748 : Removed Concept property from Program,Workflow and State domain classes.

### PR DESCRIPTION
**Pull request by samuelmale**
Work done : I removed Concept property from Program,Workflow and State domain classes and I added 'name' and 'description' .
Issue : https://issues.openmrs.org/browse/TRUNK-1748

### CODE REVIEW

**Comment by dkayiwa** [comment]
@samuelmale did you see these merge conflicts?

**Comment by samuelmale** [comment]
@dkayiwa let me look try to fix them

**Comment by samuelmale** [comment]
@dkayiwa  I have tried to resolve some. But I wonder whats causing ProgramWorkflow.java and WorkflowCollectionEditor.java to conflict?

**Comment by samuelmale** [comment]
@dkayiwa  Have resolved some conflicts. How can I get the details of the following conflicting files. 
ProgramWorkflow.java and WorkflowCollectionEditor.java

**Comment by samuelmale** [comment]
Hi @dkayiwa I'm trying to fix these merge conflicts but have first googled around it. 
I noted that the possible cause may be coz my working branch isn't upto date with upstream. I have thought this coz I did some slight changes on this branch tried to push it to origin but got this error.
`$ git push origin TRUNK-1748
Username for 'https://github.com': samuelmale
To https://github.com/samuelmale/openmrs-core.git
 ! [rejected]            TRUNK-1748 -> TRUNK-1748 (fetch first)
error: failed to push some refs to 'https://github.com/samuelmale/openmrs-core.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

`
What is confusing me is that before I branched from the master I first did
`git pull --rebase upstream master` and my master was upto date. Can you please give me a piece of advise?

**Comment by dkayiwa** [comment]
@samuelmale if you fail to fix the merge conflicts, you can as well just create brand new branch and add your changes therein.

**Comment by samuelmale** [comment]
Thanks for the advise @dkayiwa Otherwise I had given up .

**Comment by samuelmale** [comment]
@dkayiwa  Thanks. the merge conflicts are now resolved.

**Comment by samuelmale** [comment]
@dkayiwa  I have a Thought. If we merge this, it means dependant modules that used property 'concept' will be affected. I think I should leave the concept property and its getters and setters methods, and its methods in the DAO layer but just deprecate them. Does this make any sense?

**Comment by samuelmale** [comment]
@dkayiwa hi, its long time since we last looked into this, any advise towards making my commit better and fixing this issue if at all its worth :) ?

**Comment by samuelmale** [comment]
@dkayiwa fixed broken tests :)

**Comment by teleivo** [comment]
@dkayiwa and @mogoodrich could you look into this, this PR is already open quite some time now.

**Comment by dkayiwa** [comment]
@teleivo as per @mogoodrich's last comment on the ticket, i would be hesitant to merge this without the data migration changes.

File: api/src/main/java/org/openmrs/propertyeditor/WorkflowCollectionEditor.java [hunk 1]
```diff
@@ -1,140 +0,0 @@
-/**
- * This Source Code Form is subject to the terms of the Mozilla Public License,
- * v. 2.0. If a copy of the MPL was not distributed with this file, You can
- * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
- * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
- *
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/propertyeditor/WorkflowCollectionEditor.java]: Is removing this file part of the work for this ticket?

File: api/src/test/java/org/openmrs/propertyeditor/WorkflowCollectionEditorTest.java [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-
-/**
- * Tests {@link WorkflowCollectionEditor}
- */
-public class WorkflowCollectionEditorTest extends BaseContextSensitiveTest {
-	
-	/**
-	 * @see WorkflowCollectionEditor#setAsText(String)
-	 */
-	@Test
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/propertyeditor/WorkflowCollectionEditorTest.java]: Is removing this test class part of the work for this ticket?

**Comment by samuelmale** [comment]
Lets close this PR for now, I think I will work on it when I get more time @dkayiwa

**Pull request closed without merging**